### PR TITLE
Upgrade xliff-tasks to net6.0

### PR DESCRIPTION
**Pull request by jonfortescue**
While I was here, I decided to go with the analyzers and make a bunch of minor code style edits to bring the projects into alignment with C# 10.0 standards.

Part of https://github.com/dotnet/arcade/issues/8998.

### CODE REVIEW

File: src/XliffTasks.Tests/XlfDocumentTests.cs [hunk 1]
```diff
@@ -608,27 +608,27 @@ public void ValidationReportsErrorsOnMissingSourceElement()
 
         private static List<XmlSchemaException> GetValidationErrors(XlfDocument document)
         {
-            var validationErrors = new List<XmlSchemaException>();
-            Action<XmlSchemaException> exceptionHandler = e => validationErrors.Add(e);
+            List<XmlSchemaException> validationErrors = new();
+            void exceptionHandler(XmlSchemaException e) => validationErrors.Add(e);
```
Comment (garath) [inline src/XliffTasks.Tests/XlfDocumentTests.cs:612]: Fancy local function expression you got here.

File: src/XliffTasks/Model/XamlRuleDocument.cs [hunk 2]
```diff
@@ -65,9 +65,9 @@ private static string GenerateIdForPropertyMetadata(XElement element)
 
         private static string? GetComment(XElement element, string attributeName)
```
Comment (garath) [inline src/XliffTasks/Model/XamlRuleDocument.cs:66]: Nullables get complicated when moving from 3.1 to 6. Did you have any weird messages or errors like that?

Comment (jonfortescue) [reply src/XliffTasks/Model/XamlRuleDocument.cs:66]: I think the "nullables" feature is something you have to explicitly turn on in 6 and I opted not to. In any case, no weird messages or errors.

File: src/XliffTasks/StringExtensions.cs [hunk 3]
```diff
@@ -42,9 +43,9 @@ public static int GetReplacementCount(this string text)
         {
             int replacementCount = 0;
 
-            foreach (Match placeholder in s_placeholderRegex.Matches(text))
+            foreach (Match placeholder in s_placeholderRegex.Matches(text).Cast<Match>())
```
Comment (garath) [inline src/XliffTasks/StringExtensions.cs:46]: Why is the `.Cast<>()` needed? What type was being returned instead?

Comment (jonfortescue) [reply src/XliffTasks/StringExtensions.cs:46]: Actually, that's interesting now that you point it out. I was just blindly following the analyzer -- lemme dig into it.

Comment (jonfortescue) [reply src/XliffTasks/StringExtensions.cs:46]: Apparently it returns `object`

Comment (jonfortescue) [reply src/XliffTasks/StringExtensions.cs:46]: ah, I see. `.Matches` returns a `MatchCollection`, whose members are implicitly cast by the `foreach` into `Match`s but are apparently `object`s under the covers. Interesting.

File: src/XliffTasks/XliffTasks.csproj [hunk 4]
```diff
@@ -26,7 +26,7 @@
     <InternalsVisibleTo Include="XliffTasks.Tests" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
+  <ItemGroup Condition="'$(TargetFramework)' == 'net6.0'">
     <PackageReference Include="Microsoft.Build.Framework" Version="15.1.1012" PrivateAssets="All" />
     <PackageReference Include="Microsoft.Build.Utilities.Core" Version="15.1.1012" PrivateAssets="All" />
     <PackageReference Include="Microsoft.NETCore.App" Version="2.0.3" PrivateAssets="All" />
```
Comment (garath) [inline src/XliffTasks/XliffTasks.csproj:32]: Is all this still needed? `Microsoft.NETCore.App.2.0.3` is marked "[deprecated](https://www.nuget.org/packages/Microsoft.NETCore.App/2.0.3)" so it should at least get an upgrade here.

File: src/XliffTasks/StringExtensions.cs (hunk 3 above)
Comment (garath) [reply src/XliffTasks/StringExtensions.cs:46]: 😕

Comment (jonfortescue) [reply src/XliffTasks/StringExtensions.cs:46]: I can remove the cast if it makes you sad haha

Comment (garath) [reply src/XliffTasks/StringExtensions.cs:46]: Nah I trust ya

File: src/XliffTasks/XliffTasks.csproj (hunk 4 above)
Comment (jonfortescue) [reply src/XliffTasks/XliffTasks.csproj]: Upgraded the package.

**Review by garath — APPROVED** [review]

**Pull request merged**